tools/BuildAssets/psModules/CodeGenerationModules/AutoRestCodeGenerationModule/AutoRestCodeGenerationModule.psm1

```diff
@@ -0,0 +1,482 @@
+    - into the directory path provided
+
+.PARAMETER ResourceProvider
+    The Resource provider for whom to generate the code; also helps determine path where config file is located in repo
+
+.PARAMETER Version
+    The AutoRest version to use to generate the code, "latest" is recommended
+
+.PARAMETER LocalRestSpecPath
+    The local Rest Spec config file for which to generate the sdk
+
+.PARAMETER SdkRootDirectory
+    The root path in csharp-sdks-folder in config file where to generate the code
+
+.PARAMETER SdkGenerationDirectory
+    The path that where the code will be generate (overrides output-folder specified in config file)
+    
+.PARAMETER Namespace
+    The C# namespace for sdk to generate
+
+.PARAMETER ConfigFileTag
+    The tag in config file for which to generate the sdk. Code generation fails if incorrect tags are provided
+
+#>
+function Start-AutoRestCodeGenerationWithLocalRestSpec {
+    [CmdletBinding(DefaultParameterSetName="sdkRootDir")]
+    Param(
+        [Parameter(Mandatory = $true)]
+        [string] $ResourceProvider,
+
+        [Parameter(Mandatory = $false)]
+        [string] $AutoRestVersion = "latest",
+
+        [Parameter(ParameterSetName="sdkRootDir", Mandatory = $false)]
+        [string] $SdkRootDirectory,
+
+        [Parameter(ParameterSetName="sdkOutputDir", Mandatory=$false)]
+        [string] $SdkGenerationDirectory,
+
+        [Parameter(Mandatory = $false)]
+        [string] $Namespace,
+
+        [Parameter(Mandatory = $false)]
+        [string] $ConfigFileTag,
+
+        [Parameter(Mandatory = $true)]
+        [string] $LocalRestSpecPath
+    )
+    if(-not [string]::IsNullOrWhiteSpace($SdkRootDirectory)) {
+        Start-CodeGeneration -ResourceProvider $ResourceProvider -LocalRestSpecPath $LocalRestSpecPath -SdkRootDirectory $SdkRootDirectory -Namespace $Namespace -ConfigFileTag $ConfigFileTag
+    }
+    else {
+        Start-CodeGeneration -ResourceProvider $ResourceProvider -LocalRestSpecPath $LocalRestSpecPath -SdkGenerationDirectory $SdkGenerationDirectory -Namespace $Namespace -ConfigFileTag $ConfigFileTag
+    }
+}
+
+function Get-LocalSdkRepoDirectory {
+    Param(
+        [string] $currPath
+    )
+
+    while(![string]::IsNullOrEmpty($(Split-Path $currPath -Parent)))
+    {
+        if(Test-Path -Path "$currPath\src\SDKs\_metadata")
+        {
+            return $currPath
+        }
+        
+        $currPath = $(Split-Path $currPath -Parent)
+    }
+    return $currPath
+}
+
+function Start-CodeGeneration {
+    Param(
+        [string] $ResourceProvider,
+        [string] $SpecsRepoFork = "Azure",
+        [string] $SpecsRepoName = "azure-rest-api-specs",
+        [string] $SpecsRepoBranch = "master",
+        [string] $AutoRestVersion = "latest",
+        [string] $SdkRootDirectory,
+        [string] $SdkDirectory,
+        [string] $SdkGenerationDirectory,
+        [string] $Namespace,
+        [string] $ConfigFileTag,
+        [string] $LocalRestSpecPath
```

Comment:
Instead of Rest Spec, call it RestConfig
Also add a help saying this is .md file